Sources/_MatchingEngine/Regex/AST/AST.swift

```diff
@@ -88,11 +92,20 @@ extension AST {
 
   public struct Alternation: Hashable, _ASTNode {
     public let children: [AST]
+    public let pipes: [SourceLocation]
     public let location: SourceLocation
 
-    public init(_ mems: [AST], _ location: SourceLocation) {
+    public init(_ mems: [AST], pipes: [SourceLocation]) {
+      // An alternation must have at least two branches (though the branches
+      // may be empty AST nodes), and n - 1 pipes.
+      precondition(mems.count >= 2)
+      precondition(pipes.count == mems.count - 1)
+
       self.children = mems
-      self.location = location
+      self.pipes = pipes
+      self.location = SourceLocation(
+        mems.first!.location.start ..< mems.last!.location.end
```

Comment:
Could this be a computed property instead, then?

Reply:
Ah yes fair point